File: lib/src/main/scala/bruto/result/BeanExplorationResults.scala
package bruto.result

import java.util
import scala.collection.JavaConversions._

class BeanExplorationResults(var clazz: java.lang.Class[_]) {

  var methodExplorationResults: java.util.List[MethodExplorationResults] = new util.ArrayList[MethodExplorationResults]

  def add(methodExplorationResults: MethodExplorationResults) {
    this.methodExplorationResults.add(methodExplorationResults)
  }

  def getMethodExplorationResults: java.util.List[MethodExplorationResults] = {
    methodExplorationResults
  }

  def printResults(stringBuilder: java.lang.StringBuilder) = {
    stringBuilder.append("-------------- Bean Exploration Report --------------\n")
    stringBuilder.append("Class: %s\n".format(clazz.getName))
    stringBuilder.append("Testable methods: %s\n".format(methodExplorationResults.size()))

    methodExplorationResults.foreach(methodResults => methodResults.printResults(stringBuilder))

    stringBuilder.append("-------------- Bean Exploration Report End --------------\n")
  }

}

File: lib/src/main/java/bruto/result/BeanExplorationResults.java
package bruto.result;

import java.util.ArrayList;
import java.util.List;

public class BeanExplorationResults {

    private Class clazz;
    private List<MethodExplorationResults> methodExplorationResults = new ArrayList<>();

    public BeanExplorationResults(Class clazz) {
        this.clazz = clazz;
    }

    public void add(MethodExplorationResults methodExplorationResults) {
        this.methodExplorationResults.add(methodExplorationResults);
    }

    public java.util.List<MethodExplorationResults> getMethodExplorationResults() {
        return methodExplorationResults;
    }

    public void printResults(StringBuilder stringBuilder) {
        stringBuilder.append("-------------- Bean Exploration Report --------------\n");
        stringBuilder.append(String.format("Class: %s\n", clazz.getName()));
        stringBuilder.append(String.format("Testable methods: %s\n", methodExplorationResults.size()));

        for (MethodExplorationResults methodExplorationResult : methodExplorationResults) {
            methodExplorationResult.printResults(stringBuilder);
        }

        stringBuilder.append("-------------- Bean Exploration Report End --------------\n");
    }

}
